### PR TITLE
[NEW][8.0] product_variant_sale_price: Add test, fixes and translation

### PR DESCRIPTION
**Pull request by sergio-teruel**
@tafaRU @pedrobaeza Changes done.

### CODE REVIEW

File: product_variant_sale_price/__openerp__.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 {
     'name': 'Product Variant Sale Price',
-    'summary': 'Allow write fixed prices in product variants',
+    'summary': 'Allows write fixed prices in product variants',
```
Comment (pedrobaeza) [inline product_variant_sale_price/__openerp__.py]: s/Allows/Allows to

File: product_variant_sale_price/hooks.py [hunk 2]
```diff
@@ -5,9 +5,22 @@
 
 def set_sale_price_on_variant(cr, registry):
     sql = """
-            UPDATE product_product pp
-            SET fix_price = (SELECT list_price
-                                FROM product_template pt
-                                WHERE pp.product_tmpl_id = pt.id)
+        UPDATE product_product pp
```
Comment (pedrobaeza) [inline product_variant_sale_price/hooks.py:8]: You can simplify this with:

```
UPDATE product_product pp
SET fix_price = pt.list_price + (
    SELECT COALESCE(SUM(pap.price_extra), 0)
    FROM product_attribute_value_product_product_rel pav_pp_rel
    LEFT JOIN product_attribute_price pap ON
        pap.value_id = pav_pp_rel.att_id
    WHERE pav_pp_rel.prod_id = pp.id 
    AND pap.product_tmpl_id = pt.id 
)
FROM product_template pt
WHERE pt.id = pp.product_tmpl_id;
```

File: product_variant_sale_price/models/product_product.py [hunk 3]
```diff
@@ -12,8 +12,10 @@ class ProductTemplate(models.Model):
     def write(self, vals):
         res = super(ProductTemplate, self).write(vals)
         if 'list_price' in vals:
-            for variant in self.mapped('product_variant_ids'):
-                variant._onchange_lst_price()
+            for product in self:
+                product.mapped('product_variant_ids').write({
```
Comment (pedrobaeza) [inline product_variant_sale_price/models/product_product.py]: Put directly `self.mapped('product_variant_ids').write(...`

File: product_variant_sale_price/models/product_product.py [hunk 4]
```diff
@@ -32,20 +34,35 @@ def _compute_lst_price(self):
...
                 uom = product.uos_id or product.uom_id
                 vals['fix_price'] = uom._compute_price(
-                    product.uom_id.id,
-                    product.lst_price, self.env.context['uom'])
+                    product.uom_id.id, product.lst_price,
+                    self.env.context['uom'])
             else:
                 vals['fix_price'] = product.lst_price
+            if product.product_variant_count == 1:
+                    product.product_tmpl_id.list_price = vals['fix_price']
```
Comment (pedrobaeza) [inline product_variant_sale_price/models/product_product.py]: Bigger indentation

File: product_variant_sale_price/hooks.py (hunk 2 above)
Comment (pedrobaeza) [inline product_variant_sale_price/hooks.py:8]: You can check that this works making a test where you force the initial conditions on test records (perhaps making a SQL for emptying fix_price column), and then calling this method to see if it has been correctly filled. You can improve even the method to admit a keyword argument to limit the query to one record, and thus the test will be faster.

**Comment by sergio-teruel** [comment]
@tafaRU @pedrobaeza Changes done.

File: product_variant_sale_price/hooks.py [hunk 5]
```diff
@@ -3,24 +3,19 @@
...
+            FROM product_attribute_value_product_product_rel pav_pp_rel
+            LEFT JOIN product_attribute_price pap ON
+                pap.value_id = pav_pp_rel.att_id
+            WHERE pav_pp_rel.prod_id = pp.id
+            AND pap.product_tmpl_id = pt.id
+        )
+        FROM product_template pt
+        WHERE %s;
+    """ % (template_id and 'pt.id = %s' % template_id or
+           'pt.id = pp.product_tmpl_id')
```
Comment (pedrobaeza) [inline product_variant_sale_price/hooks.py:20]: You can't remove this condition when there is a `template_id` variable

Comment (pedrobaeza) [inline product_variant_sale_price/hooks.py:20]: You have to put:

```
('pt.id = %s AND ' % template_id if template_id else '' +
 'pt.id = pp.product_tmpl_id')
```

**Comment by sergio-teruel** [comment]
@pedrobaeza changes done.

**Comment by pedrobaeza** [comment]
:+1:

**Comment by tafaRU** [comment]
Thank you so much, @sergio-teruel, @pedrobaeza!

**Pull request merged**